tests/grammar.rs

```diff
@@ -48,11 +55,28 @@ mod tests {
             <rule-char>      ::= <letter> | <digit> | \"-\"
             <EOL>            ::= \"\n\"";
 
-        let grammar = bnf::Grammar::from_str(bnf_for_bnf);
-        assert!(grammar.is_ok(), "{:?} should be Ok", grammar);
-        let sentence = grammar.unwrap().generate();
-        assert!(sentence.is_ok());
-        let meta_grammar = bnf::Grammar::from_str(&sentence.unwrap());
-        assert!(meta_grammar.is_ok());
+    impl Arbitrary for Meta {
+        fn arbitrary<G: Gen>(_: &mut G) -> Meta {
```

Comment:
According to the [quickcheck docs on its Arbitrary trait](https://docs.rs/quickcheck/0.4.2/quickcheck/trait.Arbitrary.html) the provided `Gen` should be used to generate the arbitrary test data. It also highlights the importance of being able to shrink the data to a minimal failing case. This implementation seems to violate both of these contraints

Reply:
ahh, that's interesting, like maybe use that to seed random?

Reply:
Yeah we really should be able to reproduce any failures. I'll look into using weak_rng so we can provide the seed

Reply:
Seems like it. And that kind of control of `generate`'s random isn't public to integration tests like this. Do we make it public via another API? Do we move this property testing within the `Grammar` module? Is it okay to ignore the "property" aspect of property based testing and only use it as a test runner than executes multiple times?

Reply:
Shrinking a `Grammar` seems involved. Seems like would have to keep the grammar in its full tree state and do some branch trimming/grafting.

Reply:
Also, does the meta BNF grammar guarantee non-infinite grammars? Because otherwise this test occasionally fails when the generated grammar is unlucky enough to blow the stack

Reply:
I think if we're seeding then we only have to shrink on the seed. Like it's enough to reproduce to know if you seed with xyz you generate a fail case

Reply:
It does seem like I'd taken for granted a step that should have property tests applied to it

Reply:
Maybe shrinking is overly complex for BNF and these tests just opt out? Once a failure, shrinking does nothing. As I read the quickcheck docs, that seems to be the default behavior, so maybe it is not important. The key point seems to be that the same `Gen` needs to give the same test case.